### PR TITLE
Dynamic sources form part 1.

### PR DESCRIPTION
**Pull request by martinpovolny**
Initial steps on making the form sources form generated from the API.

It's problematic if this makes sense because as sources will be registered for several/different apps with different requirements, different fields in the form will be needed.

At that point a more complex logic, one that counts on the different apps will be needed.

### Done

 * load the list of sources
 * generate the combo with list of sources
  * generate pages of the wizzard
  * generate proper sequence of steps/pages

### TODO (another PR)
  * move the form definition to the sources database (entity SourceType) -- first need to add it to https://github.com/martinpovolny/topological_inventory-core

### CODE REVIEW

**Comment by martinpovolny** [comment]
To fix the (now hard-coded) calculation of entry point name for the created source, I would need this functionality: https://github.com/final-form/react-final-form/issues/387

File: src/SmartComponents/ProviderPage/providerForm.js [hunk 1]
```diff
@@ -1,141 +1,138 @@
...
+            type: validatorTypes.REQUIRED
+        }]
+    }]
+});
+
+const summaryStep = () => ({
+    fields: [{
+        name: 'summary',
+        component: 'summary',
+        assignFieldProvider: true
```
Comment (Hyperkid123) [inline src/SmartComponents/ProviderPage/providerForm.js]: `assignFieldProvider` is no longer necessary. FieldProvider is passed to each component by default.

File: src/SmartComponents/ProviderPage/providerForm.js [hunk 2]
```diff
@@ -1,141 +1,138 @@
...
+                )
             }]
-        }]
-    },
-    uiSchema: {
-    //password: {'ui:widget': 'password'}
-    }
-};
+        },
+        uiSchema: {
```
Comment (Hyperkid123) [inline src/SmartComponents/ProviderPage/providerForm.js]: You can get rid of this key entirely

**Comment by martinpovolny** [comment]
@Hyperkid123 : I addressed your feedback and added the possibility to use an environment variable for setting the BASE_PATH.

I'd like to do next steps in a new PR.

**Pull request merged**